sparse/_coo/common.py

```diff
@@ -303,12 +307,20 @@ def _dot(a, b):
         return COO(coords, data, shape=out_shape, has_duplicates=False, sorted=True)
 
     if isinstance(a, COO) and isinstance(b, np.ndarray):
+        # todo: if return_type == COO: ...
         b = b.view(type=np.ndarray).T
         return _dot_coo_ndarray_type(a.dtype, b.dtype)(a.coords, a.data, b, out_shape)
 
     if isinstance(a, np.ndarray) and isinstance(b, COO):
         b = b.T
         a = a.view(type=np.ndarray)
+        if return_type == COO:
+            coords, data, dtype = _dot_ndarray_coo_type_sparse(a.dtype, b.dtype)(
+                a, b.coords, b.data, out_shape
+            )
+            coords = np.array(coords).T
+            data = np.array(data, dtype=dtype)
+            return COO(coords, data, shape=out_shape, prune=True)
```

Comment:
In relation to https://github.com/pydata/sparse/pull/349/files#r426539625, don't rely on `prune=True` here.

Reply:
I'd say your tests should pass with `has_duplicates=False, sorted=True`